pulp_container/app/tasks/sign.py

```diff
@@ -81,20 +100,23 @@ def create_signature(manifest, reference, signing_service):
         pk of created ManifestSignature.
 
     """
-    with tempfile.TemporaryDirectory(dir=".") as working_directory:
+    async with semaphore:
         # download and write file for object storage
+        artifact = await sync_to_async(manifest._artifacts.get)()
         if settings.DEFAULT_FILE_STORAGE != "pulpcore.app.models.storage.FileSystem":
-            manifest_file = tempfile.NamedTemporaryFile(dir=working_directory, delete=False)
-            artifact = manifest._artifacts.get()
-            manifest_file.write(artifact.file.read())
-            manifest_file.flush()
+            async with tempfile.NamedTemporaryFile(dir=".", mode="wb", delete=False) as tf:
+                await tf.write(artifact.file.read())
```

Comment:
```suggestion
                await tf.write(await sync_to_async(artifact.file.read)())
```
This should probably not be blocking...

Reply:
Based on your latest commit, I am reverting the suggested change:
https://github.com/pulp/pulp_container/pull/1285#discussion_r1196595234

Can you sign this off?

Reply:
Trying to understand what i meant back then. I still think that django storages does not yet know about async. So "artifact.file.read" (which is probably a blocking call,) should not be called from an async context directly, but `sync_to_async` as i suggested.